packages/core/src/external/fhir/consolidated/deduplicate.ts

```diff
@@ -28,8 +30,11 @@ export async function deduplicate({
       duration,
     },
   };
+
   log(`Finished deduplication in ${duration} ms... Metrics: ${JSON.stringify(metrics)}`);
+  if (postHogApiKey) {
```

Comment:
Lets not call analyticsAsync in this function. Instead lets return the necessary values we need to send the event outside of this